src/services/sdk.ts

```diff
@@ -1,29 +1,183 @@
+};
+
+export const setupProvider = (): void => {
+  const global: any = window;
+  const eth = global.ethereum;
+
+  if (!eth) {
+    throw new Error(
+      "Could not create provider, because ethereum has not been set"
+    );
+  }
+
+  const provider = new providers.Web3Provider(eth);
+  setConfig({
+    provider: provider,
+    signer: provider.getSigner(),
+    store: new Store(),
+  });
+};
+
+const buildPublication = (batchData: BatchFileData): Publication => {
+  return {
+    announcementType: core.announcements.AnnouncementType.Broadcast,
+    fileUrl: batchData.url.toString(),
+    fileHash: "0x" + batchData.hash,
+  };
+};
+
+const dispatchFeedItem = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement,
+  activityPub: ActivityPub,
+  blockNumber: number
+) => {
+  const decoder = new TextDecoder();
+
+  dispatch(
+    addFeedItem({
+      fromAddress: decoder.decode((message.fromId as any) as Uint8Array),
+      blockNumber: blockNumber,
+      hash: decoder.decode((message.contentHash as any) as Uint8Array),
+      timestamp: new Date().getTime(),
+      uri: decoder.decode((message.url as any) as Uint8Array),
+      content: {
+        "@context": "https://www.w3.org/ns/activitystreams",
+        type: "Note",
+        actor: "actor",
+        content: activityPub.content || "",
+      },
+    })
+  );
+};
+
+const handleBatchAnnouncement = (dispatch: Dispatch) => (
+  announcement: BatchPublicationCallbackArgs
+) => {
+  core.batch
+    .openURL((announcement.fileUrl.toString() as any) as URL)
+    .then((reader: any) =>
+      core.batch.readFile(reader, (announcementRow: AnnouncementType) => {
+        const message = (announcementRow as unknown) as BroadcastAnnouncement;
+        const decoder = new TextDecoder();
+
+        const url = decoder.decode((message.url as any) as Uint8Array);
+        fetch(url)
+          .then((res) => res.json())
+          .then((activityPub) =>
+            dispatchFeedItem(
+              dispatch,
+              message,
+              activityPub,
+              announcement.blockNumber
+            )
+          )
+          .catch((err) => console.log(err));
+      })
+    );
+};
+
+const storeActivityPub = async (content: ActivityPub): Promise<string> => {
+  const hash = keccak256(core.activityPub.serialize(content));
+
+  await fetch(
+    `${process.env.REACT_APP_UPLOAD_HOST}/upload?filename=${encodeURIComponent(
+      hash + ".json"
+    )}`,
+    {
+      method: "POST",
+      mode: "cors",
+      body: JSON.stringify(content),
+    }
+  );
+  return hash;
+};
+
+const buildAndSignAnnouncement = async (
+  hash: string
+): Promise<SignedBroadcastAnnouncement> => {
+  return {
+    signature: "0x00000000", // TODO: call out to wallet to get this signed
+    announcementType: AnnouncementType.Broadcast,
+    contentHash: hash,
+    fromId: "0x1111", // TODO: disambiguate fromID from public key and set here
+    url: `${process.env.REACT_APP_UPLOAD_HOST}/${hash}.json`,
+  };
```

Comment:
Why not use `broadcast()` here? Or at least `createBroadcast()`?

Reply:
broadcast requires a valid dsnpID which is going to be tough to work with when the example client doesn't have registration functionality. I did swtich to use `createBroadcast`.